src/Microsoft.VisualStudio.Threading/ThreadingTools.cs

```diff
@@ -110,6 +110,30 @@ public static SpecializedSyncContext Apply(this SynchronizationContext syncConte
             return SpecializedSyncContext.Apply(syncContext, checkForChangesOnRevert);
         }
 
+        /// <summary>
+        /// Wraps a task with one that will complete as cancelled based on a cancellation token,
+        /// allowing someone to await a task but be able to break out early by cancelling the token.
+        /// </summary>
+        /// <param name="task">The task to wrap.</param>
+        /// <param name="cancellationToken">The token that can be canceled to break out of the await.</param>
+        /// <returns>The wrapping task.</returns>
+        internal static Task WithCancellationJtfAware(this Task task, CancellationToken cancellationToken)
```

Comment:
💭 This method seems to be for a more specific case than just "JTF Aware" - namely, cases where the caller is in the process of joining the result. It would recommend the following overload instead of a method with a new name:

```csharp
internal static Task WithCancellation(this Task task, boolean continueOnCapturedContext, CancellationToken cancellationToken)
```

Another alternative is it seems possible to write `WithCancellationSlow` using explicit synchronous `ContinueWith` operations, removing the need to special case the behavior depending on context.